src/www/app/persistent_server.ts

```diff
@@ -58,15 +68,25 @@ export class PersistentServerRepository implements ServerRepository {
     if (alreadyAddedServer) {
       throw new ServerAlreadyAdded(alreadyAddedServer);
     }
-    if (!OutlineServer.isServerCipherSupported(serverConfig.method)) {
-      throw new ShadowsocksUnsupportedCipher(serverConfig.method || 'unknown');
+    if (!isServerCipherSupported(serverConfig)) {
```

Comment:
I think we need to fetch the config from the ConfigSource here, so we can validate it and make sure it works.